returnn/forward.py

```diff
@@ -0,0 +1,204 @@
+            self.out_default_hdf = self.out_hdf_files["output.hdf"]
+
+        self.rqmt = {
+            "gpu": 1 if device == "gpu" else 0,
+            "cpu": cpu_rqmt,
+            "mem": mem_rqmt,
+            "time": time_rqmt,
+        }
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", resume="run", rqmt=self.rqmt)
+
+    def create_files(self):
+        config = self.create_returnn_config(
+            model_checkpoint=self.model_checkpoint,
+            returnn_config=self.returnn_config,
+            eval_mode=self.eval_mode,
+            log_verbosity=self.log_verbosity,
+            device=self.device,
+        )
+        config.write(self.out_returnn_config_file.get_path())
+
+        cmd = [
+            tk.uncached_path(self.returnn_python_exe),
+            os.path.join(tk.uncached_path(self.returnn_root), "rnn.py"),
+            self.out_returnn_config_file.get_path(),
+        ]
+        util.create_executable("rnn.sh", cmd)
+
+        # check here if model actually exists
+        assert os.path.exists(
+            self.model_checkpoint.index_path.get_path()
+        ), "Provided model does not exists: %s" % str(self.model_checkpoint)
+
+    def run(self):
+        # run everything in a TempDir as writing HDFs can cause heavy load
+        with tempfile.TemporaryDirectory(prefix=gs.TMP_PREFIX) as d:
+            print("using temp-dir: %s" % d)
+            call = [
+                tk.uncached_path(self.returnn_python_exe),
+                os.path.join(tk.uncached_path(self.returnn_root), "rnn.py"),
+                self.out_returnn_config_file.get_path(),
+            ]
+
+            try:
+                sp.check_call(call, cwd=d)
+            except Exception as e:
+                print("Run crashed - copy temporary work folder as 'crash_dir'")
+                shutil.copytree(d, "crash_dir")
+                raise e
+
+            # move log and tensorboard
+            shutil.move(os.path.join(d, "returnn.log"), "returnn.log")
+            tensorboard_dirs = glob.glob(os.path.join(d, "eval-*"))
+            for dir in tensorboard_dirs:
+                shutil.move(dir, os.path.basename(dir))
+
+            # move hdf outputs to output folder
+            for k, v in self.out_hdf_files.items():
+                shutil.move(os.path.join(d, k), v.get_path())
+
+    @classmethod
+    def create_returnn_config(
+        cls,
+        model_checkpoint,
+        returnn_config,
+        eval_mode,
+        log_verbosity,
+        device,
+        **kwargs,
+    ):
+        """
+
+        :param Checkpoint model_checkpoint:
+        :param ReturnnConfig returnn_config:
+        :param int log_verbosity:
+        :param str device:
+        :param kwargs:
+        :return:
+        """
+        assert device in ["gpu", "cpu"]
+        assert "task" not in returnn_config.config
+        assert "task" not in returnn_config.post_config
```

Comment:
```suggestion
        assert "task" not in returnn_config.config
        assert "load" not in returnn_config.config
```
The post config will anyway be checked via `check_consistency`. I propose to also check for parameter "load"